### PR TITLE
Android aarch64 cross

### PR DESCRIPTION
**Pull request by compnerd**
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

### CODE REVIEW

File: cmake/modules/SwiftConfigureSDK.cmake [hunk 1]
```diff
@@ -207,17 +180,44 @@ macro(configure_sdk_unix
...
+      if("${arch}" STREQUAL "armv7")
+        set(SWIFT_SDK_ANDROID_ARCH_${arch}_NDK_TRIPLE "arm-linux-androideabi")
+        set(SWIFT_SDK_ANDROID_ARCH_${arch}_ALT_SPELLING "arm")
+        set(SWIFT_SDK_ANDROID_ARCH_${arch}_PATH "${SWIFT_ANDROID_NDK_PATH}/platforms/android-${SWIFT_ANDROID_API_LEVEL}/arch-arm")
+        set(SWIFT_SDK_ANDROID_ARCH_${arch}_TRIPLE "armv7-none-linux-androideabi")
+      elseif("${arch}" STREQUAL "aarch64")
+        set(SWIFT_SDK_ANDROID_ARCH_${arch}_NDK_TRIPLE "aarch64-linux-android")
+        set(SWIFT_SDK_ANDROID_ARCH_${arch}_ALT_SPELLING "aarch64")
+        set(SWIFT_SDK_ANDROID_ARCH_${arch}_PATH "${SWIFT_ANDROID_NDK_PATH}/platforms/android-${SWIFT_ANDROID_API_LEVEL}/arch-arm64")
+        set(SWIFT_SDK_ANDROID_ARCH_${arch}_TRIPLE "aarch64-unknown-linux-android")
```
Comment (drodriguez) [inline cmake/modules/SwiftConfigureSDK.cmake]: `unknown` or `none`? For consistency with ARMv7, and to not confuse people later, they should be the same (the documentation uses `none`).

File: cmake/modules/SwiftConfigureSDK.cmake [hunk 2]
```diff
@@ -207,17 +180,43 @@ macro(configure_sdk_unix
...
+      else()
+        message(FATAL_ERROR "unkonwn arch for android SDK: ${arch}")
```
Comment (drodriguez) [inline cmake/modules/SwiftConfigureSDK.cmake]: Small typo in `unkonwn` (my typo, sorry).

Comment (compnerd) [reply cmake/modules/SwiftConfigureSDK.cmake]: lol, will fix :-)

File: cmake/modules/SwiftConfigureSDK.cmake (hunk 1 above)
Comment (compnerd) [reply cmake/modules/SwiftConfigureSDK.cmake]: Technically, the vendor should be `unknown`, not `none`.

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
CC: @shahmishal @zayass @gonzalolarralde @colemancda

**Comment by shahmishal** [comment]
@swift-ci build toolchain

**Comment by compnerd** [comment]
@Rostepher you may care too :-)

**Review by shahmishal — APPROVED** [review]
LGTM! cc/ @zisko @Rostepher

**Comment by shahmishal** [comment]
@swift-ci build toolchain

**Pull request merged**